pkg/distributor/distributor.go

```diff
@@ -2472,10 +2482,10 @@ func (d *Distributor) UserStats(ctx context.Context, countMethod cardinality.Cou
 		// so regardless the number of zones we have it's behaving like multi-zone is always enabled.
 		isMultiZone := d.cfg.IngestStorageConfig.Enabled || replicationSet.ZoneCount() > 1
 
-		totalStats.IngestionRate += approximateFromZones(isMultiZone, d.ingestersRing.ReplicationFactor(), zoneIngestionRate)
-		totalStats.APIIngestionRate += approximateFromZones(isMultiZone, d.ingestersRing.ReplicationFactor(), zoneAPIIngestionRate)
-		totalStats.RuleIngestionRate += approximateFromZones(isMultiZone, d.ingestersRing.ReplicationFactor(), zoneRuleIngestionRate)
-		totalStats.NumSeries += approximateFromZones(isMultiZone, d.ingestersRing.ReplicationFactor(), zoneNumSeries)
+		totalStats.IngestionRate += approximateFromZones(isMultiZone, replicationSet.ZoneCount(), d.ingestersRing.ReplicationFactor(), zoneIngestionRate)
```

Comment:
When the experimental ingest storage in enabled, I think the value computed here is wrong because what we really want is the "max per zone". I reused `approximateFromZones()` for simplicity, and I think the math was correct before this PR, but maybe at this point it's better if we have a different "approximate" function for the ingest storage.

When ingest storage is in use, a replication set is the set of ingesters for 1 partition. Given each set (which means given each partition) we want to get the max value among them. That's all.